### PR TITLE
Fix Scala Native for testing libraries

### PR DESCRIPTION
**Pull request by lolgab**
- Add missing test-runner in Scala Native tools

### CODE REVIEW

**Comment by lolgab** [comment]
Last PR I did, I broke the Scala Native build for testing libraries.. That's because Scala Native, starting from version 0.3.8, has a separate artifact for running tests, `test-runner`. Upgrading I didn't add it.
Made also a cleanup in the `build.sc` file.

File: build.sc [hunk 1]
```diff
@@ -488,15 +480,15 @@ object scalanativelib extends MillModule {
...
     def moduleDeps = Seq(scalanativelib.api)
     def ivyDeps = Agg(
-      ivy"org.scala-native::tools:${scalaNativeVersion()}",
-      ivy"org.scala-native::util:${scalaNativeVersion()}",
-      ivy"org.scala-native::nir:${scalaNativeVersion()}",
-      ivy"org.scala-native::test-runner:${scalaNativeVersion()}",
+      ivy"org.scala-native::tools:$scalaNativeVersion",
+      ivy"org.scala-native::util:$scalaNativeVersion",
+      ivy"org.scala-native::nir:$scalaNativeVersion",
+      ivy"org.scala-native::test-runner:$scalaNativeVersion",
```
Comment (lefou) [inline build.sc]: Can you move these concrete deps up into the `Deps` object?

File: scalanativelib/src/ScalaNativeModule.scala [hunk 2]
```diff
@@ -50,7 +50,8 @@ trait ScalaNativeModule extends ScalaModule { outer =>
     Seq(
       ivy"org.scala-native:tools_2.12:${scalaNativeVersion()}",
       ivy"org.scala-native:util_2.12:${scalaNativeVersion()}",
-      ivy"org.scala-native:nir_2.12:${scalaNativeVersion()}"
+      ivy"org.scala-native:nir_2.12:${scalaNativeVersion()}",
+      ivy"org.scala-native:test-runner_2.12:${scalaNativeVersion()}"
```
Comment (lefou) [inline scalanativelib/src/ScalaNativeModule.scala]: It looks like this "test" dependency ends up in a non-test project? My assumption would be to add a test dep only in test projects.

File: build.sc (hunk 1 above)
Comment (lolgab) [reply build.sc]: Sure :)

File: scalanativelib/src/ScalaNativeModule.scala (hunk 2 above)
Comment (lolgab) [reply scalanativelib/src/ScalaNativeModule.scala]: These are the dependencies only for linking (they are not Scala Native dependencies but Scala JVM 2.12 ones). Even in Scala JS there are test dependencies in the LinkerClasspath:
https://github.com/lihaoyi/mill/blob/c7873b6f2d977324f215209854a1f415e2882622/scalajslib/src/ScalaJSModule.scala#L40

File: build.sc (hunk 1 above)
Comment (lolgab) [reply build.sc]: Done ✅

File: scalanativelib/src/ScalaNativeModule.scala (hunk 2 above)
Comment (lefou) [reply scalanativelib/src/ScalaNativeModule.scala]: Thanks for clarification.

**Comment by lefou** [comment]
Thanks!

**Comment by lolgab** [comment]
> Thanks!

Thanks to you for having reviewed this so fast! :)
You are a great maintainer!

**Pull request merged**